test/benchmarks/run.sh

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+
+#------------------------------------------------------------------------------
+# Bash script to run optimizer performance tests.
+# ------------------------------------------------------------------------------
+# This file is part of solidity.
+#
+# solidity is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# solidity is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with solidity.  If not, see <http://www.gnu.org/licenses/>
+#
+# (c) 2022 solidity contributors.
+#------------------------------------------------------------------------------
+
+set -euo pipefail
+
+REPO_ROOT=$(cd "$(dirname "$0")/../../" && pwd)
+SOLIDITY_BUILD_DIR=${SOLIDITY_BUILD_DIR:-${REPO_ROOT}/build}
+
+solc="${SOLIDITY_BUILD_DIR}/solc/solc"
+benchmarks_dir="${REPO_ROOT}/test/benchmarks"
+chains_sol="${benchmarks_dir}/chains.sol"
+time_bin_path=$(which time)
+
+solc_command_legacy=("${solc}" --optimize --bin "${chains_sol}")
+solc_command_via_ir=("${solc}" --via-ir --optimize --bin "${chains_sol}")
+
+result_legacy_file=$(mktemp -t benchmark-legacy-XXXXXX.txt)
+result_via_ir_file=$(mktemp -t benchmark-via-ir-XXXXXX.txt)
+
+${time_bin_path} -o "${result_legacy_file}" --format "%e" "${solc_command_legacy[@]}" >/dev/null 2>/dev/null
+${time_bin_path} -o "${result_via_ir_file}" --format "%e" "${solc_command_via_ir[@]}" >/dev/null 2>/dev/null
```

Comment:
Ignoring stdout should be enough. There are not supposed to trigger errors, are they?

```suggestion
${time_bin_path} -o "${result_legacy_file}" --format "%e" "${solc_command_legacy[@]}" >/dev/null
${time_bin_path} -o "${result_via_ir_file}" --format "%e" "${solc_command_via_ir[@]}" >/dev/null
```

The compilation does print warnings, but we could easily fix that by adding a version pragma and an SPDX comment to the contract.

Reply:
```
Warning: This is a pre-release compiler version, please do not use it in production.
```
How about this one?

Reply:
I mean, it doesn't really affect the execution in any way, since the timing is output into the temporary file - it's just that I'd prefer for this to be quiet.

Reply:
Damn. There must always be something in the way :)

I'm suggesting this mostly because in case of an error, the first thing you want to see is the compiler output. Normally stderr should be clean so I try never to suppress it in scripts unless there's just no way around it.

Reply:
With the license and version pragma, we're getting minimal stderr output now, so I'm fine leaving it as is. Will be especially useful if compiling actually fails.